JoyboyCommunity/src/components/modal/index.tsx

```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+import {Portal} from 'react-native-portalize';
+
+import Line from '../../../assets/svgs/svgComponents/Line';
+import {
+  ButtonText,
+  Container,
+  Content,
+  IconContainer,
+  ModalContainer,
+  Overlay,
+  StyledButton,
+  Title,
+} from './styled';
+
+interface Button {
+  label: string;
+  type: 'dangerous' | 'primary' | 'secondary';
+  onPress: () => void;
+}
+interface CustomModalProps {
+  name: string;
+  buttons: Button[];
+  icon: React.ReactNode;
+  description: string;
+  visible: boolean;
+  // onClose: () => void;
+  color?: string;
+}
+const Modal: React.FC<CustomModalProps> = ({name, buttons, icon, description, visible, color}) => {
+  if (!visible) return null;
+  return (
+    <Portal>
+      <Overlay>
+        <ModalContainer>
+          <Container>
+            <Line />
+            <IconContainer>{icon}</IconContainer>
+            <Title>{name}</Title>
+            <Content>{description}</Content>
+          </Container>
+          <Container>
+            {buttons.map((button, index) => (
+              <StyledButton key={index} type={button.type} onPress={button.onPress}>
+                <ButtonText type={button.type}>{button.label}</ButtonText>
+              </StyledButton>
```

Comment:
We already have a component for buttons, [`Button`](https://github.com/keep-starknet-strange/joyboy/blob/5264c5680e32db71bf8b18f5d491d972925ab64f/JoyboyCommunity/src/components/button/index.tsx) should be used instead of using a custom one.

Reply:
As I've stated before, please use the Button component instead of using custom one.